client/src/client.ts

```diff
@@ -50,12 +62,14 @@ export class TokenServiceClient {
   public readonly jwt: string | undefined;
   public readonly env: EnvironmentName;
   public readonly serviceUrl: string;
+  public readonly fetch: (input: RequestInfo, init?: RequestInit | undefined) => Promise<Response>;
 
-  constructor (options: TokenServiceClientOptions) {
+  constructor (options: TokenServiceClientOptions | TokenServiceClientNodeOptions) {
     const {jwt, serviceUrl} = options;
     this.jwt = jwt;
     this.env = options.env || getEnv();
     this.serviceUrl = getServiceUrlFromQueryString() || serviceUrl || getServiceUrlFromEnv(this.env) || serviceUrls.production;
+    this.fetch = ("fetch" in options) ? options.fetch : fetch;
   }
```

Comment:
I think it would be good to throw an exception here if `this.fetch === undefined`.  This will make debugging easier for future users.  The exception could be something like:

```
new Error("fetch not found in options or window object")
```

Reply:
:+1: Sounds good, added.